server/build/index.js

```diff
@@ -8,7 +8,7 @@ import md5File from 'md5-file/promise'
 
 export default async function build (dir, conf = null) {
   const config = getConfig(dir, conf)
-  const buildId = uuid.v4()
+  const buildId = process.env.BUILD_ID || uuid.v4()
```

Comment:
Lets do it like this:

```
let buildId
if(typeof config.generateBuildId === 'function') {
  buildId = await config.generateBuildId() // `await` so that it can be an async function
} else {
  buildId = uuid.v4()
}
```

Reply:
Lets also add an integration test for this case.